PLUGINS.md

```diff
@@ -0,0 +1,180 @@
+# OpenShift Console Plugins
+
+Openshift Console Plugins extend the functionality of the console. Plugins are enabled by
+adding them as `dependencies` of the [console-app](frontend/packages/console-app) package.
```

Comment:
Many important things could be covered here:

- plugins are _static_ meaning their code gets bundled ("baked") into Console build output
  - plugins are _not_ loaded dynamically at runtime, which in turn simplifies the architecture (no need to sandbox plugin execution or its interaction with Console app or Bridge server)
  - `console-app` dependencies control the set of plugins to include in Console build, this can be overridden via `CONSOLE_PLUGINS` env. variable
- plugins have their code represented as Console monorepo packages
  - this increases the level of trust between Console app and its plugins
  - plugin code close to Console code avoids cross-repo PR maintenance issues
  - each plugin has an `OWNERS` file denoting its ownership (see [here](https://github.com/kubernetes/test-infra/blob/master/prow/plugins/approve/approvers/README.md) for details on reviewers and approvers)
- plugins are _declarative_ meaning they provide a list of extensions to be interpreted
  - each extension has a unique `type` and `properties` representing its parameters (data and/or callbacks)
  - (upcoming change) each extension has `flags` object to support gating by Console feature flags
- Console demo plugin is used to demonstrate specific extensions
  - it should contain at least one instance of every extension type (sadly, this isn't the case yet)
  - it should be a "synthetic" (foo-bar, not-a-real-world) example acting as a starting point for further exploration